GameFighter/Battle.cs

```diff
@@ -28,5 +29,54 @@ public static bool Fight(Warrior warrior1, Warrior warrior2)
             }
             return false;
         }
+
+        public static bool Fight(Army army1, Army army2)
+        {
+            if (army1 == null || army2 == null)
+            {
+                throw new ArgumentNullException("army1 and army2 must be initialised!");
+            }
+
+            if (army1.CountUnits == 0 || army2.CountUnits == 0)
+            {
+                throw new ArgumentException($"{army1} and {army2} must have units!");
+            }
+
+            if (!army1.IsSomeUnitAlive || !army2.IsSomeUnitAlive)
+            {
+                throw new ArgumentException($"{army1} and {army2} must have alive units!");
+            }
+
+            for (int i = 0, j = 0; i < army1.CountUnits && j <= army2.CountUnits;)
```

Comment:
Does this implementation look understandable? Try to think about the logic which doesn't depends on indexes

Reply:
I can do without indexes, but this implementation is the fastest one...